controller/controller.go

```diff
@@ -645,3 +666,14 @@ func unsetLoggerCtxValue(ctx context.Context, key string) context.Context {
 
 	return ctx
 }
+
+func matchLabels(sourceLabels, targetLabels map[string]string) bool {
+	for k, v := range sourceLabels {
+		if targetValue, ok := targetLabels[k]; !ok {
+			return false
+		} else if targetValue != v {
+			return false
+		}
+	}
+	return true
```

Comment:
If `c.matchLabels` are empty, it doesn't mean to filter any CRD or k8s resource.